pkg/virtualization/network/switchport/ethernetportallocationsettingdata.go

```diff
@@ -41,6 +45,32 @@ func (epas *EthernetPortAllocationSettingData) GetEthernetSwitchPortProfileSetti
 	return NewEthernetSwitchPortProfileSettingData(tmp)
 }
 
+func (epas *EthernetPortAllocationSettingData) GetEthernetSwitchPortProfileSettingDataForVendor(vendorGuid string) (*EthernetSwitchPortProfileSettingData, error) {
+
+	col, err := epas.GetAssociated("Msvm_EthernetPortSettingDataComponent", "Msvm_EthernetSwitchPortProfileSettingData", "", "")
+
+	if err != nil {
+		return nil, err
+	}
+	defer col.Close()
+	wmiVendorId := fmt.Sprintf("{%s}", vendorGuid)
```

Comment:
[nitpick] the code executes faster if you do the following as opposed to using sprintf: "{" + vendorGuid + "}"

Reply:
thanks for the tip.